compiler/common-artifacts/CMakeLists.txt

```diff
@@ -2,6 +2,12 @@
 find_package(PythonInterp 3.8 QUIET)
 find_package(PythonLibs 3.8 QUIET)
```

Comment:
Deprecated since version 3.12.
https://cmake.org/cmake/help/latest/module/FindPythonLibs.html

Reply:
Are you going to remove above these two lines? if so, please add a TODO.
If you have a full change, please share it with draft PR.

Reply:
Unfortunately, FindPython3 was introduced in version 3.12. So, to support python variables, these two lines should be kept.
https://cmake.org/cmake/help/latest/module/FindPython3.html

Reply:
@ragmani , you've commented to these two lines
```
find_package(PythonInterp 3.8 QUIET)
find_package(PythonLibs 3.8 QUIET)
```
with `Deprecated since version 3.12.` . 
I don't understand what the relation between these.
Why did you mention `Deprecated ` ?

Reply:
`FindPythonInterp` and `PythonLibs` are deprecated since the version 3.12 of `cmake`. So variables of these modules like `PYTHON_EXECUTABLE` or `PYTHON_VERSION_MINOR` are empty since the version 3.12 of `cmake`.
`FindPython3` is introduced since the version 3.12 of `cmake`.  We should use `FindPython3` module to get the informatio of variables since the version 3.12 of `cmake`.